### ISSUE TITLE
NonLinearOptimizer throw exception when performing Jacobian multiplication

### ISSUE THREAD

**Issue by kuriel07** [tum-vision/online_photometric_calibration#7]
hi, i'm integrating the Online Photometric Calibration into DSO with realtime camera (auto exposure one), after several capture the system crashed during evfOptimization, i tried to debug program and it's seems crashed during Jacobian multiplication after transpose jacobian (type assertion failed), the Jacobian input size is [0x0] and type =CV_64F but the resulting transpose type is CV_8U, i wonder if this cause the multiplication error, also both Jacobian and Jacobian_T size is [0x0], i wonder what cause this

note : when running the program with dataset, it works normally fine without loading loading times.txt and pcalib.txt, so the online photometric calibration is working

**Comment by NikolausDemmel** [comment]
I don't know enough about the internals to help. Maybe @PaulBergmann does.

Otherwise, I suggest to investigate what causes the 0 size matrix.

**Comment by kuriel07** [comment]
yes, after backtracing the program, i found out that points_to_optimize->size() is zero, therefore no residual data passed onto jacobian matrix, i don't know if this is the right solution, i check whether for not to perform jacobian multiplication by checking size of points_to_optimize, if the size is zero, then i skip the optimization process, right now i'm still backtracing what caused the number of points to optimize to have zero value
